sphinx/writers/latex.py

```diff
@@ -863,26 +867,30 @@ def visit_desc_parameterlist(self, node: Element) -> None:
         if self.has_tp_list:
             if self.orphan_tp_list:
                 # close type parameters list (#2)
-                self.body.append('}{')
+                self.body.append('}' + CR + '{')
                 # empty parameters list argument (#3)
                 return
         else:
             # close name argument (#1), open parameters list argument (#2)
-            self.body.append('}{')
+            self.body.append('}' + CR + '{')
         self._visit_sig_parameter_list(node, addnodes.desc_parameter)
 
     def depart_desc_parameterlist(self, node: Element) -> None:
         # close parameterlist, open return annotation
-        self.body.append('}{')
+        self.body.append('}' + CR + '{')
```

Comment:
Could you add `assert not self.orphan_tp_list`? I would like to be sure that the algorithm is correct. If the assertion fails, then something else is wrong and I need to think more.

Reply:
Actually, I also believe that the check `if self.orphan_tp_list:` in `visit_desc_parameterlist` is not needed. I wonder why I added it... because if `orphan_tp_list` is True, then the the next sibling would not be a `desc_parameterlist`, meaning that we would never call this visitor. But whether we have a type parameters list or not (`self.has_tp_list`), we should close the previous macro argument and open a new one. Therefore, I think we should instead replace in `visit_desc_parameterlist`

```py
        if self.has_tp_list:
            if self.orphan_tp_list:
                # close type parameters list (#2)
                self.body.append('}{')
                # empty parameters list argument (#3)
                return
        else:
            # close name argument (#1), open parameters list argument (#2)
            self.body.append('}{')
```

by

```py
        if self.has_tp_list:
		    assert not self.orphan_tp_list
        # case 1: close name argument (#1), open parameters list argument (#2)
		# case 2: close type parameters list (#2), open parameters list argument (#3)
        self.body.append('}{')
```

Reply:
I will commit your suggestion... no urgency here don't worry we can wait for when you will have time.

Reply:
(Just check that it doesn't break existing tests... I need to remember how I first imagined the visitors + macros but I'll do it tomorrow)

Reply:
You almost fooled me with a tab but I am super cautious :smiley:

Reply:
Hmm sorry this is breaking the case of
```
.. py:function:: generic_ret[R]() -> R
```
my bad! I was wrong... and you were right when you first wrote the code.